src/main/kotlin/org/openrewrite/RecipeOrigin.kt

```diff
@@ -18,27 +44,32 @@ class RecipeOrigin(
      */
     fun isFromCoreLibrary() = groupId == "org.openrewrite" && coreLibs.contains(artifactId)
 
-    fun githubUrl() =
-        if (isFromCoreLibrary()) {
-            "https://github.com/openrewrite/rewrite"
-        } else if (artifactId == "rewrite-gradle") {
-            "https://github.com/openrewrite/rewrite/tree/main/rewrite-gradle"
+    private fun convertNameToJavaPath(recipeName: String): String =
+        recipeName.replace('.', '/') + ".java"
+
+    fun githubUrl(recipeName: String, source: URI): String {
+        val sourceString = source.toString()
+
+        // YAML recipes will have a source that ends with META-INF/rewrite/something.yml
+        if (sourceString.substring(sourceString.length - 3) == "yml") {
+            val ymlPath = sourceString.substring(source.toString().lastIndexOf("META-INF"))
+            return artifactIdToBaseGithubUrl[artifactId].toString() + "/resources/" + ymlPath
         } else {
-            "https://github.com/openrewrite/$artifactId"
+            return artifactIdToBaseGithubUrl[artifactId].toString() + "/java/" + convertNameToJavaPath(recipeName)

```

Comment:
Wouldn't this lead to a 404 for any `yml` recipes?

Reply:
See 3 lines above this. Yaml recipes won't reach this step.

Reply:
Awesome; sorry, missed that reviewing in GitHub; ignore me! ;)

Reply:
Np ^_^